src/widgets/helper/ChannelView.cpp

```diff
@@ -2101,12 +2101,21 @@ void ChannelView::handleLinkClick(QMouseEvent *event, const Link &link,
                 }
             }
 
-            value.replace("{user}", layout->getMessage()->loginName)
-                .replace("{channel}", this->channel_->getName())
-                .replace("{msg-id}", layout->getMessage()->id)
-                .replace("{message}", layout->getMessage()->messageText);
-
-            value = getApp()->commands->execCommand(value, channel, false);
+            value = getApp()->commands->execCustomCommand(
+                QStringList(), Command{"(modaction)", value}, true, channel,
+                {
+                    {"user.name", layout->getMessage()->loginName},
+                    {"msg.id", layout->getMessage()->id},
+                    {"msg.text", layout->getMessage()->messageText},
+
+                    // old placehodlers
+                    {"user", layout->getMessage()->loginName},
+                    {"msg-id", layout->getMessage()->id},
+                    {"message", layout->getMessage()->messageText},
+
+                    // new version of this is inside execCustomCommand
+                    {"channel", this->channel()->getName()},
```

Comment:
Will we mark these as deprecated and remove them later or are we just gonna keep these as legacy?

Reply:
I don't like breaking things, maybe we do it some time should though.

Reply:
Yes that shouldn't be changed right now, but maybe a note for users that new variables should be used now and maybe remove this like after over a few stable releases to not keep stale legacy forever is reasonable.